xchange-bittrex/src/main/java/org/knowm/xchange/bittrex/service/BittrexDigest.java

```diff
@@ -27,10 +32,64 @@ public static BittrexDigest createInstance(String secretKeyBase64) {
   @Override
   public String digestParams(RestInvocation restInvocation) {
 
-    String invocationUrl = restInvocation.getInvocationUrl();
+    String invocationUrl = getInvocationUrl(restInvocation);
     Mac mac = getMac();
     mac.update(invocationUrl.getBytes());
 
     return String.format("%0128x", new BigInteger(1, mac.doFinal()));
   }
+
+  // copy paste methods from si.mazi.rescu.RestInvocation
+  static String getInvocationUrl(RestInvocation restInvocation) {
+    // HARDCODED API URL. Reason: evade INVALID_SIGNATURE error. Url used as digest payload
```

Comment:
when does this INVALID_SIGNATURE happen?

Reply:
Fort example if you use proxying. In this case signature creates with your proxy url, bur should be with exchange api url

Reply:
then all this code can be replaced with:
```
// define this in BinanceExchange
public static String DEFAULT_SSL_URI = "https://bittrex.com/api/"

// pass the exchangeSpecification to BittrexDigest constructor and then:
String invocationUrl = restInvocation.getInvocationUrl();
String digsetUrl = invocationUrl;
if (!invocationUrl.startsWith(DEFAULT_SSL_URI)) {
  String specUri = exchangeSpecification.getSslUri();
  digsetUrl = invocationUrl.replace(specUri, DEFAULT_SSL_URI);
}
```